### DISCUSSION TITLE
In the model Wholebody ct segmentation, can we segment seperate organs as we want ?

### DISCUSSION THREAD

**Discussion by hari3100** [Project-MONAI/model-zoo#616] [Q&A]
As Im trying to use this model to do individual organ segmentation, for eg. for lungs , liver, spleen, kidneys, etc.
Im using the model from monai , im able to run the code and everything runs fine , but I get a single nii.gz containing all the classes segmentation in it .

I followed a youtube video to do my work: [video](https://www.youtube.com/watch?v=NA8_Yi_q7X4)

I can't seem to find a way how can I only obtain the segmentation Im interested in

Below is the screenshot of one of my scans , as you can see the model did a great job at segmenting all the organs , but I want it to only segment lungs , or liver or something else . I understand maybe as of now we can directly change the number of classes the model will predict but can we atleast stop those unwanted predicted results from being saved in the final output.nii.gz file ?
![image](https://github.com/user-attachments/assets/f19f9b70-78ad-48a0-82e7-e13e688f6e27)

**Thank you for any potential help !!**

**Comment by diazandr3s** [comment]
Hi @hari3100,

Thanks for sharing this.

As this model is being trained to segment all those regions, it'll always output them. 

One thing you could do is to take only the organs you're interested in by masking the predicted mask. Say for instance, you know liver is represented by index 3, you only take the mask region with that index. 

Another thing you could consider is using the Auto3DSeg models available here: https://discourse.slicer.org/t/new-extension-monai-auto3dseg-raise-the-bar-in-ai-medical-image-segmentation/35680?u=diazandr3s

Just install the latest Slicer and install the Auto3DSeg via the Extension Manager.

There you can find the whole body CT segmentation models and other models specific to lungs, abdomen, spine, etc


Plus, you can also download TCIA datasets in Slicer - it is a great platform


Hope this help,

**Reply by hari3100** [reply]
Hi @diazandr3s , thank you for your reply and yes i'll try those things which you mentioned.

I believe my question was not put properly, as I'm not doing my entire work in 3D slicer , rather I was using it as a means to visually see my end results that came from my python code which is using this whole body ct segmentation model. 
Nevertheless , thank you for this help. 

Also, i did find a way around this problem of mine , and yes you are right as the model is trained to segment all the features that it detects in a given input image, it does saves a nii.gz file with all them, but I found that the segmentation that we process have all the labels that were detected in a given image , I simply took those and created a secondary nii.gz file which consist of just the organ(s) i need. 

Thank you once again!

**Reply by diazandr3s** [reply]
A similar discussion here: https://github.com/Project-MONAI/tutorials/issues/1724
